scripts/tools/generate_esp32_chip_factory_bin.py

```diff
@@ -430,53 +443,66 @@ def any_base_int(s): return int(s, 0)
     parser = argparse.ArgumentParser(description='Chip Factory NVS binary generator tool')
 
     # These will be used by CommissionalbeDataProvider
-    parser.add_argument('-p', '--passcode', type=any_base_int, required=True,
+    parser.add_argument('-p', '--passcode', type=any_base_int,
                         help='The setup passcode for pairing, range: 0x01-0x5F5E0FE')
-    parser.add_argument('-d', '--discriminator', type=any_base_int, required=True,
+    parser.add_argument('-d', '--discriminator', type=any_base_int,
                         help='The discriminator for pairing, range: 0x00-0x0FFF')
 
     # These will be used by DeviceAttestationCredentialsProvider
-    parser.add_argument('--dac-cert', type=str, required=True,
+    parser.add_argument('--dac-cert', type=str,
                         help='The path to the DAC certificate in der format')
-    parser.add_argument('--dac-key', type=str, required=True,
+    parser.add_argument('--dac-key', type=str,
                         help='The path to the DAC private key in der format')
-    parser.add_argument('--pai-cert', type=str, required=True,
+    parser.add_argument('--pai-cert', type=str,
                         help='The path to the PAI certificate in der format')
-    parser.add_argument('--cd', type=str, required=True,
+    parser.add_argument('--cd', type=str,
                         help='The path to the certificate declaration der format')
 
     # These will be used by DeviceInstanceInfoProvider
-    parser.add_argument('--vendor-id', type=any_base_int, required=False, help='Vendor id')
-    parser.add_argument('--vendor-name', type=str, required=False, help='Vendor name')
-    parser.add_argument('--product-id', type=any_base_int, required=False, help='Product id')
-    parser.add_argument('--product-name', type=str, required=False, help='Product name')
-    parser.add_argument('--hw-ver', type=any_base_int, required=False, help='Hardware version')
-    parser.add_argument('--hw-ver-str', type=str, required=False, help='Hardware version string')
-    parser.add_argument('--mfg-date', type=str, required=False, help='Manufacturing date in format YYYY-MM-DD')
-    parser.add_argument('--serial-num', type=str, required=False, help='Serial number')
-    parser.add_argument('--rd-id-uid', type=str, required=False,
+    parser.add_argument('--vendor-id', type=any_base_int, help='Vendor id')
+    parser.add_argument('--vendor-name', type=str, help='Vendor name')
```

Comment:
argument types are by default str, you can remove several of `type=str` arguments.

Reply:
Removed all the `type=str`